### PR TITLE
Feature/tg 84 create course

### PR DESCRIPTION
**Pull request by RamiroJGuillen**
# Summary

Created Create Course Endpoint and Tests.

# Details

* Added Create Course endpoint and updated interfaces, services and repositories.
* Removed Code property from Course Model and created a new DTO for Course creation.
* Updated Context without Code property.
* Updated Tests.
* Updated Migrations.
* Updated the branch with latest develop changes.
* Added repository and services test

# Evidence
Screenshot 2024-02-02 at 13 30 43
Screenshot 2024-02-02 at 13 37 31
Screenshot 2024-02-02 at 13 37 59
Screenshot 2024-02-02 at 13 38 53

### CODE REVIEW

**Comment by AJN-jalaU** [comment]
please add some images as evidence. You can take this PR as an example:
https://github.com/JU-DEV-LatamBootcamp/Klockan-Backend/pull/12

**Review by AJN-jalaU — APPROVED** [review]

File: KlockanAPI/src/KlockanAPI.Application/DTOs/Course/CourseDto.cs [hunk 1]
```diff
@@ -1,10 +1,9 @@
 ﻿namespace KlockanAPI.Application.DTOs.Course;
 
-public class CourseDto
+public class CourseDTO
 {
     public int Id { get; set; }
     public string Name { get; set; } = string.Empty;
-    public string Code { get; set; } = string.Empty;
     public string Description { get; set; } = string.Empty;
```
Comment (Adrian-Mendoza-Jala) [inline KlockanAPI/src/KlockanAPI.Application/DTOs/Course/CourseDto.cs]: Description is Optional

File: KlockanAPI/src/KlockanAPI.Presentation/Program.cs [hunk 2]
```diff
@@ -68,6 +68,10 @@ public static WebApplicationBuilder ConfigureServicesAndMiddlewares(WebApplicati
         builder.Services.AddFluentValidationAutoValidation();
         builder.Services.AddFluentValidationClientsideAdapters();
 
+        // ***********  FLUENT VALIDATION ************
```
Comment (Adrian-Mendoza-Jala) [inline KlockanAPI/src/KlockanAPI.Presentation/Program.cs]: This part is repeating

**Review by Adrian-Mendoza-Jala — APPROVED** [review]

File: KlockanAPI/src/KlockanAPI.Presentation/Program.cs (hunk 2 above)
Comment (RamiroJGuillen) [reply KlockanAPI/src/KlockanAPI.Presentation/Program.cs]: solved

File: KlockanAPI/src/KlockanAPI.Application/DTOs/Course/CourseDto.cs (hunk 1 above)
Comment (RamiroJGuillen) [reply KlockanAPI/src/KlockanAPI.Application/DTOs/Course/CourseDto.cs]: solved!

**Pull request merged**